lib/rubocop/cop/rspec/repeated_description.rb

```diff
@@ -0,0 +1,82 @@
+module RuboCop
+  module Cop
+    module RSpec
+      # Check for repeated description strings in example groups.
+      #
+      # @example
+      #
+      #     # bad
+      #     RSpec.describe User do
+      #       it 'is valid' do
+      #         # ...
+      #       end
+      #
+      #       it 'is valid' do
+      #         # ...
+      #       end
+      #     end
+      #
+      #     # good
+      #     RSpec.describe User do
+      #       it 'is valid when first and last name are present' do
+      #         # ...
+      #       end
+      #
+      #       it 'is valid when last name only is present' do
+      #         # ...
+      #       end
+      #     end
+      #
+      class RepeatedDescription < Cop
+        include RuboCop::Cop::ConfigurableEnforcedStyle
+
+        def_node_matcher :example, <<-PATTERN
+          (block $(send _ #{Examples::ALL.node_pattern_union} ...) ...)
+        PATTERN
+
+        # @!method scope_change?(node)
+        #
+        #   Detect if the node is an example group or shared example
+        #
+        #   Selectors which indicate that indicate we should stop searching
+        #
+        def_node_matcher :scope_change?,
+                         (ExampleGroups::ALL + SharedGroups::ALL).block_pattern
+
+        MSG = 'Repeated description'.freeze
+
+        def on_block(node)
+          return unless example_group?(node)
+
+          repeated_descriptions(node).each do |repeated_hook|
+            add_offense(repeated_hook, :expression)
+          end
+        end
+
+        private
+
+        def repeated_descriptions(node)
+          hooks_in_group(node)
+            .group_by(&:method_args)
+            .reject { |args, _| args.empty? }
+            .values
+            .reject(&:one?)
+            .flatten
```

Comment:
Would this be more readable using local variables?

```rb
hooks_with_arguments = hooks_in_group(node).group_by(&:method_args).reject { |args, _| args.empty? }
hooks_with_identical_arguments = hooks_with_arguments.values.reject(&:one?)
hooks_with_identical_arguments.flatten
```

If I got the names wrong, it’s because I didn’t understand the code.